xfrm_state_linux.go

```diff
@@ -213,7 +213,9 @@ func (h *Handle) xfrmStateGetOrDelete(state *XfrmState, nlProto int) (*XfrmState
 		req.AddData(out)
 	}
 	if state.Src != nil {
-		out := nl.NewRtAttr(nl.XFRMA_SRCADDR, state.Src)
+		sAddr := nl.XfrmAddress{}
+		sAddr.FromIP(state.Src)
+		out := nl.NewRtAttr(nl.XFRMA_SRCADDR, sAddr.ToIP())
```

Comment:
Wanted to reuse the `nl.XfrmAddress` structure. But if you are ok with it, I can simply just change this line to:

```
out := nl.NewRtAttr(nl.XFRMA_SRCADDR, state.Src.To16())
```